StudenSchool_DO/HW2/WorkWithUser.cs

```diff
@@ -0,0 +1,44 @@
+﻿namespace HW2;
+
+public delegate bool CheckingCondition(int value);
+
+internal static class WorkWithUser
+{
+    public static int GetNumberFromUser(CheckingCondition IsCondition)
+    {
+        int numberOfLines;
+
+        while (!IsValid(out numberOfLines, IsCondition))
+        {
+            DesignedMenu.WriteServiceMessages("Неправильный ввод. Повторите");
+        }
+
+        return numberOfLines;
+    }
+
+    private static bool IsValid(out int value, CheckingCondition IsCondition)
```

Comment:
`out` params usually are the last arguments in order. e.g. private static bool IsValid(CheckingCondition IsCondition, out int value)